src/PackageVersions/FallbackVersions.php

```diff
@@ -0,0 +1,78 @@
+<?php
+
+namespace PackageVersions;
+
+/**
+ * @internal
+ *
+ * This is a fallback for {@see \PackageVersions\Versions::getVersion()}
+ * Do not use this class directly: it is intended to be only used when
+ * {@see \PackageVersions\Versions} fails to be generated, which typically
+ * happens when running composer with `--no-scripts` flag)
+ */
+final class FallbackVersions
+{
+    private function __construct()
+    {
+    }
+
+    /**
+     * @param string $packageName
+     *
+     * @return string
+     *
+     * @throws \OutOfBoundsException if a version cannot be located
+     * @throws \UnexpectedValueException if the composer.lock file could not be located
+     */
+    public static function getVersion(string $packageName) : string
+    {
+        $versions = iterator_to_array(self::getVersions(self::getComposerLockPath()));
+
+        if (! array_key_exists($packageName, $versions)) {
+            throw new \OutOfBoundsException(
+                'Required package "' . $packageName . '" is not installed: cannot detect its version'
+            );
+        }
+
+        return $versions[$packageName];
+    }
+
+    /**
+     * @return string
+     *
+     * @throws \UnexpectedValueException
+     */
+    private static function getComposerLockPath() : string
+    {
+        // bold assumption, but there's not here to fix everyone's problems.
+        $checkedPaths = [__DIR__ . '/../../../../../composer.lock', __DIR__ . '/../../composer.lock'];
```

Comment:
@Ocramius Perhaps some `$path = dirname(__DIR__, $level++) . '/composer.lock'`-based check as a better chance alternative? Of course with some reasonable threshold for `$level` increment.